.github/workflows/mod-check.yaml

```diff
@@ -0,0 +1,36 @@
+name: Mod check
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+      - release-*
+  
+  pull_request:
+    branches:
+      - main
+      - release-*
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
```

Comment:
as it's mostly around linter, adding it to https://github.com/ceph/ceph-csi-operator/blob/main/.github/workflows/lint.yaml makes more sense, WDYT?

Reply:
done. and removed the mod-check workflow